### PR TITLE
Event Subscription using aggregate

### PR DESCRIPTION
**Pull request by chauberahul1993**

### CODE REVIEW

File: lib-persistence-manager/persistencemgr/redis.go [hunk 1]
```diff
@@ -1411,3 +1411,95 @@ func (p *ConnPool) TTL(table, key string) (int, *errors.Error) {
 	}
 	return time, nil
 }
+
+// CreateAggregateSubscriptionIndex is used to create and save secondary index
+/* CreateAggregateSubscriptionIndex take the following keys are input:
+1. index is the name of the index to be created
+2. key is for the index
+*/
+func (p *ConnPool) CreateAggregateSubscriptionIndex(index, aggregateID string, hostIP []string) error {
```
Comment (jeevan-kamkar) [inline lib-persistence-manager/persistencemgr/redis.go:1420]: Rename this function it shouldn't be subscription index as we discussed

File: svc-events/evmodel/events.go [hunk 2]
```diff
@@ -566,3 +593,45 @@ func DeleteUndeliveredEventsFlag(destination string) error {
 	}
 	return nil
 }
+
+// SaveAggregateSubscription is to save subscription details of device
+func SaveAggregateSubscription(aggregateID string, hostIP []string) error {
```
Comment (jeevan-kamkar) [inline svc-events/evmodel/events.go:598]: As discussed this should be done in Aggregation Service during the creation of aggregate, not during event subscription please   update the same

**Review by jeevan-kamkar — CHANGES_REQUESTED** [review]
Address the requested changes

File: lib-persistence-manager/persistencemgr/redis.go (hunk 1 above)
Comment (chauberahul1993) [reply lib-persistence-manager/persistencemgr/redis.go]: done

File: svc-events/evmodel/events.go (hunk 2 above)
Comment (chauberahul1993) [reply svc-events/evmodel/events.go:598]: done

**Review by jeevan-kamkar — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**